conans/util/config_parser.py

```diff
@@ -44,6 +44,10 @@ def __init__(self, text, allowed_fields=None, strip_comments=False):
                 if self._allowed_fields and field not in self._allowed_fields:
                     raise ConanException("ConfigParser: Unrecognized field '%s'" % field)
                 current_lines = []
+                # Duplicated section
+                if field in self._sections:
+                    raise ConanException(f"ConfigParser: Duplicated section: [{field}]. It could"
+                                         f"cause some unexpected behavior.")
```

Comment:
Not anymore right? It could not cause new unexpected behaviour as we are disallowing it :)